cmake/detect-arch.cmake

```diff
@@ -85,6 +85,12 @@ elseif("${ARCH}" MATCHES "parisc")
 elseif("${ARCH}" MATCHES "rs6000")
     set(BASEARCH "rs6000")
     set(BASEARCH_RS6000_FOUND TRUE)
+elseif("${ARCH}" MATCHES "riscv32")
+    set(BASEARCH "riscv32")
+    set(BASEARCH_RISCV32_FOUND TRUE)
+elseif("${ARCH}" MATCHES "riscv64")
+    set(BASEARCH "riscv64")
+    set(BASEARCH_RISCV64_FOUND TRUE)
```

Comment:
These should likely have a common basearch, like `risc` or `riscv` or something like that.
Like basearch `x86` has both 32 and 64bit, and same for basearch `arm`.

Having a common basearch is useful because often the same feature/instruction-set/etc tests are common, or they can even use much of the same accelerated code.

Reply:
Should be fixed in recent force-push.